### PR TITLE
Use root Group instead of the App instance

### PR DESCRIPTION
**Pull request by azhai**
* Use root Group instead of the App instance
* Remove some methods of App when Group extends App

### CODE REVIEW

**Comment by renanbastos93** [comment]
> * Use root Group instead of the App instance
> * Remove some methods of App when Group extends App

Sorry, but I don't understand the gain of this change.
Can you explain this to us?

**Comment by azhai** [comment]
Why Group should have the same methods as App, rather than extends App?

File: req_test.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+		if sizeStr := c.FormValue("size"); sizeStr != "" {
+			size, _ = strconv.Atoi(sizeStr)
+		}
+		data := Map{"page": page, "size": size}
+		c.JSON(Map{"code": 200, "data": data})
+	})
+	return grp
+}
+
+//func CreateSeniorGroup() *Group {
```
Comment (renanbastos93) [inline req_test.go:41]: Why do you comment it?

**Comment by renanbastos93** [comment]
> Why Group should have the same methods as App, rather than extends App?

looks promising so I liked it to remove code and keeping your API, but this change requires many tests and reviews from the community in specially `maintainers`

File: req_test.go (hunk 1 above)
Comment (azhai) [reply req_test.go:41]: The test is used for another unofficial version  [fiber-u8l](https://gitee.com/azhai/fiber-u8l)
I has add some methods like ParamStr(), FetchInt() in Ctx class.

**Comment by Fenny** [comment]
Hi @azhai, thank you for your PR. It's an interesting approach, could you share a use-case for this change some can see the benefit. Thank you!

**Comment by Fenny** [comment]
I'm closing this issue because we have no plans in changing the return types for `v1`, feel free to re-open to continue the discussion with a good use-case.

Thank you for your PR!

**Pull request closed without merging**